x-pack/legacy/plugins/siem/server/lib/detection_engine/rules/update_rule_actions.ts

```diff
@@ -1,54 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License;
- * you may not use this file except in compliance with the Elastic License.
- */
-
-import {
-  AlertsClient,
-  AlertServices,
-  PartialAlert,
-} from '../../../../../../../plugins/alerting/server';
-import { getRuleActionsSavedObject } from '../rule_actions/get_rule_actions_saved_object';
-import { readRules } from './read_rules';
-import { transformRuleToAlertAction } from '../../../../common/detection_engine/transform_actions';
-
-interface UpdateRuleActions {
-  alertsClient: AlertsClient;
-  savedObjectsClient: AlertServices['savedObjectsClient'];
-  ruleAlertId: string;
-}
-
-export const updateRuleActions = async ({
-  alertsClient,
-  savedObjectsClient,
-  ruleAlertId,
-}: UpdateRuleActions): Promise<PartialAlert | null> => {
-  const rule = await readRules({ alertsClient, id: ruleAlertId });
-  if (rule == null) {
-    return null;
-  }
-
-  const ruleActions = await getRuleActionsSavedObject({
-    savedObjectsClient,
-    ruleAlertId,
-  });
-
-  if (!ruleActions) {
-    return null;
-  }
-
-  return alertsClient.update({
-    id: ruleAlertId,
-    data: {
-      actions: !ruleActions.alertThrottle
-        ? ruleActions.actions.map(transformRuleToAlertAction)
-        : [],
-      throttle: null,
-      name: rule.name,
-      tags: rule.tags,
-      schedule: rule.schedule,
-      params: rule.params,
-    },
-  });
-};
```

Comment:
NOTE: The race condition is that we cannot update a rule after it has been enabled/disabled like this. This is what causes the errors such as these below:

```ts
server    log   [12:18:35.986] [error][alerting][alerting][plugins][plugins] Executing Alert "63b828b5-24b9-4d55-83ee-8a8201fe2d76" has resulted in Error: [security_exception] missing authentication credentials for REST request [/_security/user/_has_privileges], with { header={ WWW-Authenticate={ 0="Bearer realm=\"security\"" & 1="ApiKey" & 2="Basic realm=\"security\" charset=\"UTF-8\"" } } 
```

Plus when we remove this, we remove an additional ES call as well.